server/router/ride.ts

```diff
@@ -145,6 +145,93 @@ router.put('/:id', validateUser('User'), (req, res) => {
   db.update(res, Ride, { id }, body, tableName);
 });
 
+// Create edit instances and update a repeating ride's edits field
+router.put('/:id/edits', (req, res) => {
+  const { params: { id }, body: { deleteOnly, origDate, rider, startTime, endTime,
+    startLocation, endLocation, driver },
+  } = req;
+
+  db.getById(res, Ride, id, tableName, (masterRide) => {
+    const origStartTimeOnly = moment(masterRide.startTime).format('HH:mm:ss');
+    const origStartTime = moment(`${origDate}T${origStartTimeOnly}`).toISOString();
+
+    const origEndTimeOnly = moment(masterRide.endTime).format('HH:mm:ss');
+    const origEndTime = moment(`${origDate}T${origEndTimeOnly}`).toISOString();
+
+    const masterFirstDay = moment(masterRide.startTime).format('YYYY-MM-DD');
+
+    if (origDate === masterFirstDay) {
+      // if editing the first instance, move the start/endTimes to the next occurrence
+      const weekday = moment(origDate).weekday();
+      const weekdayIndex = masterRide.recurringDays.findIndex((day: number) => day === weekday);
+
+      const nextWeekdayIndex = ((weekdayIndex + 1) % masterRide.recurringDays.length);
+      const nextWeekday = masterRide.recurringDays[nextWeekdayIndex];
+
+      const addDays = (nextWeekday > weekday) ? nextWeekday - weekday : 7 - weekday + nextWeekday;
+      const nextOccurrence = moment(origDate).add(addDays, 'days');
+
+      const newDate = nextOccurrence.format('YYYY-MM-DD');
+      const newStartTime = moment(`${newDate}T${origStartTimeOnly}`).toISOString();
+      const newEndTime = moment(`${newDate}T${origEndTimeOnly}`).toISOString();
+
+      const setNextOperation = { $SET: { startTime: newStartTime, endTime: newEndTime } };
+      db.update(res, Ride, { id }, setNextOperation, tableName);
+
+      // if deleteOnly = false, create a one-time ride with the new fields
+      if (!deleteOnly) {
+        const newRide = new Ride({
+          id: uuid(),
+          rider,
+          startLocation,
+          endLocation,
+          startTime,
+          endTime,
+          driver,
+        });
+        db.create(res, newRide);
+      }
+    } else {
+      const deleteId = uuid();
+      const deleteRide = new Ride({
+        id: deleteId,
+        rider: masterRide.rider,
+        startLocation: masterRide.startLocation,
+        endLocation: masterRide.endLocation,
```

Comment:
make sure if startLocation is a preset location, youre only passing the id into startLocation (and same for endLocation)